lib/abide_dev_utils/cem/generate/reference.rb

```diff
@@ -262,6 +264,47 @@ def generate!
             resource_reference_builder
           end
 
+          # This function act as a filter for controls based on the profile and level selections.
+          # There are few scanarios that can happen:
+          # 1. If no selections are made for profile or level, then all profiles and levels of control will be selected.
+          # 2. If selections are made for profile, then only the selected profile and all levels of control will be selected.
+          # 3. If selections are made for level, then only the selected level and all profiles of control will be selected.
+          # This function adds in some runtime overhead because we're checking each control's level and profile which is
+          # what we're going to be doing later when building the level and profile markdown, but this is
+          # necessary to ensure that the reference.md is generated the way we want it to be.
+          def verify_profile_and_level_selections
+            return true if @opts[:select_profile].nil? && @opts[:select_level].nil?
+
+            if @opts[:select_profile].nil? && !@opts[:select_level].nil?
+              @control.levels.each do |level|
+                @valid_level << level if control_level_filter(level)
```

Comment:
The filtering function filter out the levels/functions that we don't want. It basically returns a boolean whether a level/profile from `@control` is in `@opts[:select_profile/select_level]`. If that function return a `true`, then we use that level. Otherwise, we don't use that level in the generator. Sorry if the naming made this confusing.